fedimint-core/src/amount.rs

```diff
@@ -105,96 +93,58 @@ impl Amount {
         self.msats as f64 / 1000.0
     }
 
-    pub fn checked_sub(self, other: Self) -> Option<Self> {
-        Some(Self {
-            msats: self.msats.checked_sub(other.msats)?,
-        })
-    }
-
     pub fn checked_add(self, other: Self) -> Option<Self> {
         Some(Self {
             msats: self.msats.checked_add(other.msats)?,
         })
     }
-}
-
-impl std::fmt::Display for Amount {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{} msat", self.msats)
-    }
-}
-
-impl std::ops::Rem for Amount {
-    type Output = Self;
 
-    fn rem(self, rhs: Self) -> Self::Output {
+    pub fn saturating_add(self, other: Self) -> Self {
         Self {
-            msats: self.msats % rhs.msats,
+            msats: self.msats.saturating_add(other.msats),
         }
     }
-}
-
-impl std::ops::RemAssign for Amount {
-    fn rem_assign(&mut self, rhs: Self) {
-        self.msats %= rhs.msats;
-    }
-}
-
-impl std::ops::Div for Amount {
-    type Output = u64;
 
-    fn div(self, rhs: Self) -> Self::Output {
-        self.msats / rhs.msats
+    pub fn saturating_add_to(&mut self, other: Self) {
```

Comment:
This variant seems unnecessary, standard library doesn't have anything like that. A bit more verbosity is OK, IMO.